doc/guides/method.rdoc

```diff
@@ -0,0 +1,35 @@
+= Documenting a Method
+
+== Structure
+
+=== Synopsis
+
+Begin with a one-line synopsis that tells what the method does.  Example:
+  *  call-seq:
+  *    array <=> other_array -> -1, 0, or 1
+  *
+  *  Returns -1, 0, or 1 as +self+ is less than, equal to, or greater than +other_array+.
+
+== Multiple Signatures
+
+\[Is 'signatures' the correct term?]
+
+When a method has multiple signatures,
+document each signature separately, as if it were a distinct method.
+
+In <tt>array.c</tt>, for example, we have this:
+  *  call-seq:
+  *    array[index] -> object or nil
+  *    array[start, length] -> object or nil
+  *    array[range] -> object or nil
+  *    array.slice(index) -> object or nil
```

Comment:
I think we should not list all aliases as separate lines in `call-seq`.  We should pick one as the canonical method, and have the other one be documented as `alias for $canonical_method`.